### PR TITLE
Fix #429: Image left/centre side issue

### LINKED ISSUE #429: Images need to be centered in state cards

### PR DESCRIPTION
**Pull request by rt4914**
## Explanation

This PR makes sure, that images are left in options and centre in content card.
Screenshot_1574252776
Screenshot_1574252895

## Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

### CODE REVIEW

**Comment by rt4914** [comment]
@veena14cs Please check whether the `true/false` has been transferred correctly across all html parser usage or not.

True for image in centre
false for image to left.

**Comment by veena14cs** [comment]
> @veena14cs Please check whether the `true/false` has been transferred correctly across all html parser usage or not.
> 
> True for image in centre
> false for image to left.

Okay will check it through.

File: app/src/main/java/org/oppia/app/player/state/SelectionInteractionView.kt [hunk 1]
```diff
@@ -72,9 +75,10 @@ class SelectionInteractionView @JvmOverloads constructor(
           },
           bindView = { view, viewModel ->
             val binding = DataBindingUtil.findBinding<ItemSelectionInteractionItemsBinding>(view)!!
-            binding.htmlContent = htmlParserFactory.create(entityType, explorationId).parseOppiaHtml(
-              viewModel.htmlContent, binding.itemSelectionContentsTextView
-            )
+            binding.htmlContent =
+              htmlParserFactory.create(entityType, explorationId, /* imageCenterAlign= */ false).parseOppiaHtml(
```
Comment (rt4914) [inline app/src/main/java/org/oppia/app/player/state/SelectionInteractionView.kt]: @veena14cs This `false` value in this file is correct.

File: app/src/main/java/org/oppia/app/player/state/SelectionInteractionView.kt [hunk 2]
```diff
@@ -89,9 +93,10 @@ class SelectionInteractionView @JvmOverloads constructor(
           },
           bindView = { view, viewModel ->
             val binding = DataBindingUtil.findBinding<MultipleChoiceInteractionItemsBinding>(view)!!
-            binding.htmlContent = htmlParserFactory.create(entityType, explorationId).parseOppiaHtml(
-              viewModel.htmlContent, binding.multipleChoiceContentTextView
-            )
+            binding.htmlContent =
+              htmlParserFactory.create(entityType, explorationId, /* imageCenterAlign= */ false).parseOppiaHtml(
```
Comment (rt4914) [inline app/src/main/java/org/oppia/app/player/state/SelectionInteractionView.kt]: @veena14cs This `false` value in this file is correct.

File: app/src/main/java/org/oppia/app/player/state/StateFragmentPresenter.kt [hunk 3]
```diff
@@ -132,9 +132,10 @@ class StateFragmentPresenter @Inject constructor(
         },
         bindView = { view, viewModel ->
           val binding = DataBindingUtil.findBinding<ContentItemBinding>(view)!!
-          binding.htmlContent = htmlParserFactory.create(entityType, explorationId).parseOppiaHtml(
-            (viewModel as ContentViewModel).htmlContent.toString(), binding.contentTextView
-          )
+          binding.htmlContent =
+            htmlParserFactory.create(entityType, explorationId, /* imageCenterAlign= */ true).parseOppiaHtml(
```
Comment (rt4914) [inline app/src/main/java/org/oppia/app/player/state/StateFragmentPresenter.kt:136]: @veena14cs This `false` value in this file is correct.

File: app/src/main/java/org/oppia/app/player/state/StateFragmentPresenter.kt [hunk 4]
```diff
@@ -144,9 +145,10 @@ class StateFragmentPresenter @Inject constructor(
         },
         bindView = { view, viewModel ->
           val binding = DataBindingUtil.findBinding<FeedbackItemBinding>(view)!!
-          binding.htmlContent = htmlParserFactory.create(entityType, explorationId).parseOppiaHtml(
-            (viewModel as FeedbackViewModel).htmlContent.toString(), binding.feedbackTextView
-          )
+          binding.htmlContent =
+            htmlParserFactory.create(entityType, explorationId, /* imageCenterAlign= */ true).parseOppiaHtml(
```
Comment (rt4914) [inline app/src/main/java/org/oppia/app/player/state/StateFragmentPresenter.kt:149]: @veena14cs This `false` value in this file is correct.

File: app/src/main/java/org/oppia/app/testing/HtmlParserTestActivity.kt [hunk 5]
```diff
@@ -22,7 +22,7 @@ class HtmlParserTestActivity : InjectableAppCompatActivity() {
     val rawDummyString =
       "\u003cp\u003e\"Let's try one last question,\" said Mr. Baker. \"Here's a pineapple cake cut into pieces.\"\u003c/p\u003e\u003coppia-noninteractive-image alt-with-value=\"\u0026amp;quot;Pineapple cake with 7/9 having cherries.\u0026amp;quot;\" caption-with-value=\"\u0026amp;quot;\u0026amp;quot;\" filepath-with-value=\"\u0026amp;quot;pineapple_cake_height_479_width_480.png\u0026amp;quot;\"\u003e\u003c/oppia-noninteractive-image\u003e\u003cp\u003e\u00a0\u003c/p\u003e\u003cp\u003e\u003cstrong\u003eQuestion 6\u003c/strong\u003e: What fraction of the cake has big red cherries in the pineapple slices?\u003c/p\u003e"
     val htmlResult: Spannable =
-      htmlParserFactory.create( /* entityType= */ "exploration", /* entityId= */ "oppia")
+      htmlParserFactory.create( /* entityType= */ "exploration", /* entityId= */ "oppia", /* imageCenterAlign= */ false)
```
Comment (rt4914) [inline app/src/main/java/org/oppia/app/testing/HtmlParserTestActivity.kt:25]: @veena14cs Is this correct?

File: app/src/main/java/org/oppia/app/topic/overview/TopicOverviewFragmentPresenter.kt [hunk 6]
```diff
@@ -35,7 +35,7 @@ class TopicOverviewFragmentPresenter @Inject constructor(
   private val topicOverviewViewModel = getTopicOverviewViewModel()
   private lateinit var topicId: String
   private val htmlParser: HtmlParser by lazy {
-    htmlParserFactory.create(/* entityType= */ "topic", topicId)
+    htmlParserFactory.create(/* entityType= */"topic", topicId, /* imageCenterAlign= */ true)
```
Comment (rt4914) [inline app/src/main/java/org/oppia/app/topic/overview/TopicOverviewFragmentPresenter.kt:38]: @veena14cs Is this correct?

File: app/src/sharedTest/java/org/oppia/app/parser/HtmlParserTest.kt [hunk 7]
```diff
@@ -47,7 +47,7 @@ class HtmlParserTest {
   @Test
   fun testHtmlContent_handleCustomOppiaTags_parsedHtmlDisplaysStyledText() {
     val textView = activityTestRule.activity.findViewById(R.id.test_html_content_text_view) as TextView
-    val htmlParser = htmlParserFactory.create(/* entityType= */ "", /* entityId= */ "")
+    val htmlParser = htmlParserFactory.create(/* entityType= */ "", /* entityId= */ "", /* imageCenterAlign= */ false)
```
Comment (veena14cs) [inline app/src/sharedTest/java/org/oppia/app/parser/HtmlParserTest.kt]: Here it should be true. As it is content and not item selection.

File: app/src/sharedTest/java/org/oppia/app/parser/HtmlParserTest.kt [hunk 8]
```diff
@@ -65,15 +65,15 @@ class HtmlParserTest {
   @Test
   fun testHtmlContent_nonCustomOppiaTags_notParsed() {
     val textView = activityTestRule.activity.findViewById(R.id.test_html_content_text_view) as TextView
-    val htmlParser = htmlParserFactory.create(/* entityType= */ "", /* entityId= */ "")
+    val htmlParser = htmlParserFactory.create(/* entityType= */ "", /* entityId= */ "", /* imageCenterAlign= */ false)
```
Comment (veena14cs) [inline app/src/sharedTest/java/org/oppia/app/parser/HtmlParserTest.kt]: Ditto as above

**Review by veena14cs — APPROVED** [review]
LGTM

File: app/src/sharedTest/java/org/oppia/app/parser/HtmlParserTest.kt (hunk 7 above)
Comment (rt4914) [reply app/src/sharedTest/java/org/oppia/app/parser/HtmlParserTest.kt]: Done.

File: app/src/sharedTest/java/org/oppia/app/parser/HtmlParserTest.kt (hunk 8 above)
Comment (rt4914) [reply app/src/sharedTest/java/org/oppia/app/parser/HtmlParserTest.kt]: Done.

File: app/src/main/java/org/oppia/app/testing/HtmlParserTestActivity.kt (hunk 5 above)
Comment (veena14cs) [reply app/src/main/java/org/oppia/app/testing/HtmlParserTestActivity.kt:25]: it should be true here. As it is checking content.

File: app/src/main/java/org/oppia/app/topic/overview/TopicOverviewFragmentPresenter.kt (hunk 6 above)
Comment (rt4914) [reply app/src/main/java/org/oppia/app/topic/overview/TopicOverviewFragmentPresenter.kt:38]: I think this needs to be in center only.

Comment (veena14cs) [reply app/src/main/java/org/oppia/app/topic/overview/TopicOverviewFragmentPresenter.kt:38]: This is correct if here we are only showing content.

**Pull request merged**